modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/distruptor/inbound/InboundTransactionEvent.java

```diff
@@ -85,6 +112,13 @@ public InboundTransactionEvent(MessageStore.AndesTransaction transaction, Inboun
      * @throws AndesException
      */
     public void commit() throws AndesException {
+
+        if(currentBatchSize > maxBatchSize) {
+            currentBatchSize = 0;
+            throw new AndesException("Current enqueued batch size exceeds maximum transactional batch size of " +
+            maxBatchSize + " bytes." );
+        }
+
         if (log.isDebugEnabled()) {
             log.debug("Prepare for commit");
```

Comment:
unsure that would make sense. But consider this: suppose there are 4 transactions that are happening in parallel (for different publishers) and we have enabled debugs (to troubleshoot , or just for kicks!) . in such a situation seeing "Prepare for commit" 4 times would make sense? or would that improve our ability to debug further? so the more general question is how can we keep track of whose doing what transaction? when? how we are going to troubleshoot potential problems (that we may encounter?!). food for thought !